frontend/src/components/JoinProjectDialog.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useState } from "react";
+import "../styles/JoinProjectDialog.css";
+import { Token } from "../contexts/Auth.tsx";
+import call, { Method } from "../utils/api.ts";
+
+// Define an enum for error types
+enum ErrorType {
+    None = "",
+    JoinFailed = "Failed to join project. Please try again.",
+    InvalidInput = "Invalid input. Please check your entries.",
+    Success = "Successfully joined the project!",
+}
+
+interface JoinProjectDialogProps {
+    token: Token;
+}
+
+export default function JoinProjectDialog({
+    token,
+}: Readonly<JoinProjectDialogProps>) {
+    const [isDialogOpen, setIsDialogOpen] = useState(false);
+    const [projectID, setProjectID] = useState("");
+    const [error, setError] = useState<ErrorType>(ErrorType.None);
+
+    const openDialog = () => {
+        setIsDialogOpen(true);
+    };
+    const closeDialog = () => {
+        setIsDialogOpen(false);
+        setError(ErrorType.None);
+    };
+    //Function to handle project join. Take in param e for submission event
+    const handleJoin = (e: React.FormEvent) => {
+        e.preventDefault(); //prevent page refresh 
+        //POST request call
+        call(`project/${projectID}/join`, Method.Put, token)
+            .then((response) => {
+                if (!response.ok) {
+                    setError(ErrorType.JoinFailed);
+                } else {
+                    setError(ErrorType.Success);
+                }
+            })
+            .catch((err) => {
+                console.error(err);
+                setError(ErrorType.JoinFailed);
```

Comment:
What can throw? You should read the feedback I left Simon as it applies here. We should know exactly what can go wrong and explicitly handle it, rather than just giving up.